testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -3717,6 +3739,222 @@ fn cost_voting_integration() {
     channel.stop_chains_coordinator();
 }
 
+#[test]
+#[ignore]
+fn mining_events_integration_test() {
+    if env::var("BITCOIND_TEST") != Ok("1".into()) {
+        return;
+    }
+
+    let small_contract = "(define-public (f) (ok 1))".to_string();
+
+    let spender_sk = StacksPrivateKey::from_hex(SK_1).unwrap();
+    let addr = to_addr(&spender_sk);
+
+    let spender_sk_2 = StacksPrivateKey::from_hex(SK_2).unwrap();
+    let addr_2 = to_addr(&spender_sk_2);
+
+    let tx = make_contract_publish(&spender_sk, 0, 600000, "small", &small_contract);
+    let tx_2 = make_contract_publish(&spender_sk, 1, 610000, "small", &small_contract);
+    let mb_tx =
+        make_contract_publish_microblock_only(&spender_sk_2, 0, 620000, "small", &small_contract);
+
+    let (mut conf, miner_account) = neon_integration_test_conf();
+
+    conf.initial_balances.push(InitialBalance {
+        address: addr.clone().into(),
+        amount: 10000000,
+    });
+    conf.initial_balances.push(InitialBalance {
+        address: addr_2.clone().into(),
+        amount: 10000000,
+    });
+
+    conf.node.mine_microblocks = true;
+    conf.node.wait_time_for_microblocks = 30000;
+    conf.node.microblock_frequency = 1000;
+
+    conf.miner.min_tx_fee = 1;
+    conf.miner.first_attempt_time_ms = i64::max_value() as u64;
+    conf.miner.subsequent_attempt_time_ms = i64::max_value() as u64;
+
+    test_observer::spawn();
+
+    conf.events_observers.push(EventObserverConfig {
+        endpoint: format!("localhost:{}", test_observer::EVENT_OBSERVER_PORT),
+        events_keys: vec![
+            EventKeyType::AnyEvent,
+            EventKeyType::MinedBlocks,
+            EventKeyType::MinedMicroblocks,
+        ],
+    });
+
+    let mut btcd_controller = BitcoinCoreController::new(conf.clone());
+    btcd_controller
+        .start_bitcoind()
+        .map_err(|_e| ())
+        .expect("Failed starting bitcoind");
+
+    let mut btc_regtest_controller = BitcoinRegtestController::new(conf.clone(), None);
+    let http_origin = format!("http://{}", &conf.node.rpc_bind);
+
+    btc_regtest_controller.bootstrap_chain(201);
+
+    eprintln!("Chain bootstrapped...");
+
+    let mut run_loop = neon::RunLoop::new(conf);
+    let blocks_processed = run_loop.get_blocks_processed_arc();
+
+    let channel = run_loop.get_coordinator_channel().unwrap();
+
+    thread::spawn(move || run_loop.start(None, 0));
+
+    // give the run loop some time to start up!
+    wait_for_runloop(&blocks_processed);
+
+    // first block wakes up the run loop
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    // first block will hold our VRF registration
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    // second block will be the first mined Stacks block
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    let account = get_account(&http_origin, &miner_account);
```

Comment:
we copy these "get_account" checks around but i don't think it's needed for the test

Reply:
True... but I guess it doesn't hurt to ensure that the starting state of all the relevant accounts are what you expect them to be? Let me know what you think.